rest/route/keys_routes.go

```diff
@@ -146,3 +147,49 @@ func (h *keysPostHandler) Execute(ctx context.Context, sc data.Connector) (Respo
 
 	return ResponseData{}, nil
 }
+
+type keysDeleteHandler struct {
+	keyName string
+}
+
+func getKeysDeleteRouteManager(route string, version int) *RouteManager {
+	return &RouteManager{
+		Route: route,
+		Methods: []MethodHandler{
+			MethodHandler{
+				PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+				Authenticator:     &RequireUserAuthenticator{},
+				RequestHandler:    &keysDeleteHandler{},
+				MethodType:        http.MethodDelete,
+			},
+		},
+		Version: version,
+	}
+}
+
+func (h *keysDeleteHandler) Handler() RequestHandler {
+	return &keysDeleteHandler{}
+}
+
+func (h *keysDeleteHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	vars := mux.Vars(r)
+	h.keyName = vars["key_name"]
+	if strings.TrimSpace(h.keyName) == "" {
+		return &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "empty key name",
+		}
+	}
+
```

Comment:
Let's also add validation here that the key exists and 404 if not

Reply:
Validation was added, but to Execute since that's where the user object is. A thought that crosses my mind is that MustHaveUser should be in ParseAndValidate instead, and the user object stored in the handler struct. Thoughts?